addons/account/account_bank_statement.py

```diff
@@ -735,6 +742,24 @@ def process_reconciliation(self, cr, uid, id, mv_line_dicts, context=None):
                 if mv_line.reconcile_id:
                     raise osv.except_osv(_('Error!'), _('A selected move line was already reconciled.'))
 
+        # Particular use case : using reconciled move lines (fr: rapprochement bancaire)
+        num_already_reconciled_items = len([x for x in mv_line_dicts if x.get('is_reconciled', False)])
+        if num_already_reconciled_items > 0:
+            # Check only using already reconciled move lines
+            if len(mv_line_dicts) != num_already_reconciled_items:
+                raise osv.except_osv(_('Error!'), _('You cannot mix reconciled and unreconciled items.'))
+            # Check all move lines are from the same move
+            move_id = aml_obj.browse(cr, uid, mv_line_dicts[0].get('counterpart_move_line_id'), context=context).move_id.id
+            for mv_line_dict in mv_line_dicts:
+                if move_id != aml_obj.browse(cr, uid, mv_line_dict.get('counterpart_move_line_id'), context=context).move_id.id:
+                    raise osv.except_osv(_('Error!'), _('You cannot mix items from different journal entries.'))
+            # Link move lines to the bank statement
+            for mv_line_dict in mv_line_dicts:
```

Comment:
mv_line_ids  =[line_dict['counterpart_move_line_id'] for line_dict in mv_line_dicts]
aml_obj.write(cr, uid, mv_line_ids, {'statement_id': st_line.statement_id.id}, context=context)